sql/settype_test.go

```diff
@@ -0,0 +1,176 @@
+package sql
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSetCompare(t *testing.T) {
+	tests := []struct {
+		vals []string
+		collation Collation
+		val1 interface{}
+		val2 interface{}
+		expectedCmp int
+	}{
+		{[]string{"one", "two"}, Collation_Default, nil, 1, -1},
```

Comment:
What happens for other kinds of comparisons (other than strings and ints)?

Also you don't test comparison behavior for invalid lookups, e.g. "three"